crates/isograph_lang_parser/src/parse_iso_literal.rs

```diff
@@ -19,11 +22,44 @@ use crate::{
     ParseResultWithLocation, ParseResultWithSpan, PeekableLexer,
 };
 
-pub fn parse_iso_fetch(
-    iso_fetch_text: &str,
+pub enum IsoLiteralExtractionResult {
+    ClientFieldDeclaration(WithSpan<ResolverDeclaration>),
+    EntrypointDeclaration(WithSpan<EntrypointTypeAndField>),
+}
+
+pub fn parse_iso_literal(
+    iso_literal_text: &str,
+    definition_file_path: ResolverDefinitionPath,
+    text_source: TextSource,
+) -> Result<WithSpan<IsoLiteralExtractionResult>, WithLocation<IsographLiteralParseError>> {
+    let mut tokens = PeekableLexer::new(iso_literal_text);
+    tokens
+        .with_span(|tokens| {
+            let discriminator: WithSpan<StringKey> = tokens
+                .parse_string_key_type(IsographLangTokenKind::Identifier)
+                .map_err(|with_span| with_span.map(IsographLiteralParseError::from))
+                .map_err(|err| err.to_with_location(text_source))?;
+            match discriminator.item.lookup() {
+                "entrypoint" => Ok(IsoLiteralExtractionResult::EntrypointDeclaration(
+                    parse_iso_fetch(tokens, text_source)?,
+                )),
+                "field" => Ok(IsoLiteralExtractionResult::ClientFieldDeclaration(
+                    parse_iso_client_field(tokens, definition_file_path, text_source)
+                        .map_err(|err| err.1)?,
+                )),
+                _ => Err(WithLocation::new(
+                    IsographLiteralParseError::LeftoverTokens,
+                    Location::new(text_source, discriminator.span),
+                )),
```

Comment:
let's make this a new type of `IsographLiteralParseError`, e.g. `ExpectedFieldOrEntrypoint`. There's no need to print the token that we actually got, as the error (when printed) will highlight the incorrect token

Reply:
Fixed with [`e6dc985` (#4)](https://github.com/isographlabs/isograph/pull/4/commits/e6dc98574269a8b312507993bd74f960ef36620b)